lib/ansible/plugins/lookup/cpm_status.py

```diff
@@ -0,0 +1,156 @@
+# (c) 2018, Western Telematic Inc. <kenp@wti.com>
+# (c) 2012-18 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = """
+lookup: cpm_status
+author: "Western Telematic Inc. (@wtinetworkgear)"
+version_added: "2.7"
+short_description: Get status and parameters from WTI OOB and PDU devices
+description:
+    - "Get various status and parameters from WTI OOB and PDU devices"
+options:
+  _terms:
+    description:
+      - This is the Action to send the module.
+    required: true
+    choices: [ "temperature", "firmware", "status", "alarms" ]
+  cpm_url:
+    description:
+      - This is the URL of the WTI device to send the module.
+    required: true
+  cpm_username:
+    description:
+      - This is the Basic Authentication Username of the WTI device to send the module.
+    required: true
+  cpm_password:
+    description:
+      - This is the Basic Authentication Password of the WTI device to send the module.
+    required: true
+  use_https:
+    description:
+      - Designates to use an https connection or http connection.
+    required: false
+    type: bool
+    default: true
+  validate_certs:
+    description:
+      - If false, SSL certificates will not be validated. This should only be used
+        on personally controlled sites using self-signed certificates.
+    required: false
+    type: bool
+    default: true
+  use_proxy:
+    description: Flag to control if the lookup will observe HTTP proxy environment variables when present.
+    type: boolean
+    default: True
+"""
+
+EXAMPLES = """
+# Get temperature
+  - name: run Get Device Temperature
+    debug: msg="{{ lookup('cpm_status',
```

Comment:
hi @samdoran , absolutely, could you point us to an lookup example that we could pattern this after ? I am having a little trouble

Reply:
The example I gave is a pretty good. I don't think we have any examples of using the lookup across multiple lines. There are quite a few examples that need to me updated to use YAML rather than `k=v`, which is confusing.

You can look at [`consul_kv.py`](https://github.com/ansible/ansible/blob/3e96c99b074cd92788c307370d31a70049201574/lib/ansible/plugins/lookup/consul_kv.py#L55-L55) or [`onepassword.py`](https://github.com/ansible/ansible/blob/3e96c99b074cd92788c307370d31a70049201574/lib/ansible/plugins/lookup/onepassword.py#L52-L52) for examples.